src/coreclr/jit/instr.cpp

```diff
@@ -1096,19 +1193,68 @@ void CodeGen::inst_RV_TT_IV(instruction ins, emitAttr attr, regNumber reg1, GenT
     }
 }
 
+//------------------------------------------------------------------------
+// IsEmbeddedBroadcastEnabled: determine if embedded broadcast can be enabled
+//
+// Arguments:
+//    ins       -- The instruction being emitted
+//    op        -- The second operand of the instruction.
+//
+#if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
+bool CodeGenInterface::IsEmbeddedBroadcastEnabled(instruction ins, GenTree* op)
+{
+    if (!GetEmitter()->UseEvexEncoding())
+    {
+        return false;
+    }
+    // need to check if the datatype is EB compatible, say 32-, 64-bit.
+    insFlags flags                    = instInfo[ins];
+    bool     IsEmbBroadcastCompatible = (flags & INS_Flags_EmbeddedBroadcastSupported) != 0;
```

Comment:
This should probably have its own helper API, as many of the other flag checks do.